### PR TITLE
fix(no-dupe-keys): handle nested objects & getter and setter with the same name

### PR DESCRIPTION
**Pull request by magurotuna**
This is a part of #330 

What I did In this PR:

- adding more tests (mostly from ESLint)
- allowing it to handle nested objects e.g. `const obj = { a: { b: 1, b: 2 } };`
- adding proper handling for getter and setter with the same name e.g. `const obj = { get a() {}, set a() {} };`

### CODE REVIEW

**Comment by magurotuna** [comment]
@bartlomieju ready for review. PTAL

File: src/rules/no_dupe_keys.rs [hunk 1]
```diff
@@ -102,60 +212,111 @@ mod tests {
...
+    assert_lint_err::<NoDupeKeys>(r#"var x = { a: b, ['a']: b };"#, 8);
+    assert_lint_err::<NoDupeKeys>(r#"var x = { '': 1, '': 2 };"#, 8);
+    assert_lint_err::<NoDupeKeys>(r#"var x = { '': 1, [``]: 2 };"#, 8);
+    assert_lint_err::<NoDupeKeys>(r#"var x = { 012: 1, 10: 2 };"#, 8);
+    assert_lint_err::<NoDupeKeys>(r#"var x = { 0b1: 1, 1: 2 };"#, 8);
+    assert_lint_err::<NoDupeKeys>(r#"var x = { 0o1: 1, 1: 2 };"#, 8);
+    // TODO(magurotuna): this leads to panic due to swc error
+    // It seems like tsc v4.0.2 cannot handle this either
+    // playground: https://www.typescriptlang.org/play?target=99&ts=4.0.2#code/MYewdgzgLgBCBGArGBeGBvAUDGBGMAXDACwBMANJgL4DcQA
+    // assert_lint_err::<NoDupeKeys>(r#"var x = { 1n: 1, 1: 2 };"#, 8);
```
Comment (bartlomieju) [inline src/rules/no_dupe_keys.rs:286]: @magurotuna could you report that in SWC repo?

**Review by bartlomieju — APPROVED** [review]
LGTM, thanks for the fix @magurotuna

File: src/rules/no_dupe_keys.rs (hunk 1 above)
Comment (magurotuna) [reply src/rules/no_dupe_keys.rs:286]: @bartlomieju I thought I would do so, but as I commented, the TypeScript compiler can't parse this either. So I felt like it's okay for now. But I'll report it anyway 😄

Comment (magurotuna) [reply src/rules/no_dupe_keys.rs:286]: https://github.com/swc-project/swc/issues/1173

Filed an issue about it

**Pull request merged**